extensions/azdata/src/azdata.ts

```diff
@@ -134,21 +135,106 @@ export async function upgradeAzdata(outputChannel: vscode.OutputChannel): Promis
 }
 
 /**
- * Checks whether a newer version of azdata is available - and if it is prompts the user to download and
- * install it.
+ * Checks whether azdata is installed - and if it is not then invokes the process of azdata installation.
+ * @param outputChannel Channel used to display diagnostic information
+ * @param userRequested true means that this operation by was requested by a user by executing an ads command.
+ */
+export async function checkAndInstallAzdata(outputChannel: vscode.OutputChannel, userRequested: boolean = false): Promise<IAzdataTool | undefined> {
+	try {
+		const azdata = await findAzdata(outputChannel); // find currently installed Azdata
+		// Don't block on this since we want the extension to finish activating without needing user input
+		checkAndUpgradeAzdata(azdata, outputChannel, userRequested).catch(err => {
+			vscode.window.showWarningMessage(loc.updateError(err));
+			outputChannel.appendLine(loc.updateError(err));
+		}); //update if available and user wants it.
+		return findAzdata(outputChannel); // now again find and return the currently installed azdata
```

Comment:
Yeah, let me handle properly in a follow-up task, for now, I think I should just block on the checkAndUpgradeAzdata call here, would that be ok?

Reply:
No because that will delay extension activation which will cause ADS to behave improperly if you keep the call in activate.

I'd just do what I suggested and make the calls all blocking but in activate use .then to actually update the version. That should be pretty straightforward to do and is likely how we actually want to handle this.

Reply:
Done.